File: src/gestorAplicacion/Restaurantes/Restaurante.java
package gestorAplicacion.Restaurantes;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

public class Restaurante implements Serializable {

    private String nombre;
    private List<Mesa> mesas = new ArrayList<>();
    private int precio;
    private int id;
    private Menu menu;

    public Restaurante(String nombre, int precio, int id) {
        this.nombre = nombre;
        this.precio = precio;
        this.id = id;
        this.menu = new Menu();
        inicializar();
    }

    public void añadirMesa(Mesa mesa) {
        mesas.add(mesa);
    }

    public String mostrarMesas() {
        StringBuilder s = new StringBuilder();
        for (Mesa mesa : mesas) {
            s.append(mesa.toString());
        }
        return s.toString();
    }

    @Override
    public String toString() {
        return nombre + ", precio: " + precio;
    }

    public void inicializar() {
        for (int i = 1; i < 8; i++) {
            this.añadirMesa(new Mesa(i));
        }
    }

    public Menu getMenu() {
        return menu;
    }

    public void setMenu(Menu menu) {
        this.menu = menu;
    }
    //*Getters y setters */
    
    public List<Mesa> getMesas() {
        return mesas;
    }

    public void setMesas(List<Mesa> mesas) {
        this.mesas = mesas;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public int getPrecio() {
        return precio;
    }

    public void setPrecio(int precio) {
        this.precio = precio;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    
}

File: python/src/gestor_aplicacion/restaurantes/Restaurante.py
from gestor_aplicacion.restaurantes import Mesa
from enum import Enum

class Restaurante(Enum):
    
    RESTAURANTE_MEXICANO = (1, 50000)
    RESTAURANTE_ITALIANO = (2, 25000)
    RESTAURANTE_GOURMET = (3, 30000)
    RESTAURANTE_VEGETARIANO = (4, 15000)

    def __init__(self, idRestaurante, precioRestaurante):
        self.idRestaurante = idRestaurante
        self.precioRestaurante = precioRestaurante
        
        
    def añadirMesa(self, mesa):
        self.mesas.append(mesa)

    def mostrarMesas(self):
        s = ""
        for mesa in self.mesas:
            s += str(mesa)
        return s

    @staticmethod
    def buscarPorId(idRestaurante):
        for restaurante in Restaurante:
            if int(restaurante.getIdRestaurante()) == int(idRestaurante):
                return restaurante

        print("no encontrado")
        return None

    @staticmethod
    def calcularPrecioRestaurantes(idsRestaurantes):
        precioTotal = 0
        for idRestaurante in idsRestaurantes:
            restaurante = Restaurante.buscarPorId(idRestaurante)
            if restaurante is not None:
                precioTotal += restaurante.getPrecioResetaurante()
        return precioTotal

    def inicializar(self):
        for i in range(1, 8):
            self.añadirMesa(Mesa(i))

    # Getters y setters
    def getMesas(self):
        return self.mesas

    def setMesas(self, mesas):
        self.mesas = mesas

    def getPrecioResetaurante(self):
        return self.precioRestaurante
    
    def getIdRestaurante(self):
        return self.idRestaurante
